### PR TITLE
Dockerize Services & App

### PR DESCRIPTION
**Pull request by alex-stout**
This PR eases the burden of installation of the key services using Docker.

Here is the itemized list of changes:
- Added a `docker-compose.yml` file into the `/api` folder.
	- I added it here because it would not work properly with the `.env` file when placed in the root of the project. 
- Added a `/api/init_sql/` directory with the `scout.sql` script in there. 
	- This was added because the Docker script needs to have a directory to mount to the container for the MySQL service. What this does is installs the Scout db/schema into the DB automatically. If I find a better way to do this in the future so that there isn't two files, I'll create a new PR, but for now that's the only way I could see how to do it.
- Updated a the docs to include instructions on getting the Docker services up and running.
- Updated the `installer.rb` to ask if the user is using a Docker install for local development. If so, it doesn't go through the steps of asking them for the SQL and NoSQL configuration questions. 
	- I made sure to maintain the same position & variable names for the `.env` file. You'll notice the `docker-compose.yml` file references them. This is handy so you don't have to rename the variables to fit into the required naming convention of the Docker images.  

In the future I will add Dockerization for the production install. There's going to have to be a discussion on whether or not that is something that should be done. There's more to consider with this approach for that use case.

*Note: I wasn't able to find the issue reference for this.*

### CODE REVIEW

**Comment by alex-stout** [comment]
Just updated to add persistence to the databases. Now on initial startup, volumes will be mounted to `api/data-volumes` which will contain the data for both the SQL and NoSQL database. This allows for the services to be shutdown and re-run without data loss.

**Comment by battlecow** [comment]
@alex-stout Should exclude the package-lock file from this set of changes.

**Comment by alex-stout** [comment]
@battlecow 	I've removed the `package-lock` file from the change list. I've also added a `Dockerfile` and `.dockerignore` for the app itself (not the services).

**Comment by battlecow** [comment]
My thought here is if we are going to include a Dockerfile we should at a minimum run it as non-root user since node is added upstream.  Added a gist of my thoughts: https://gist.github.com/battlecow/7d98a64088e9ed3a2fb9283700a19fea
If the container is meant for dev the NODE_ENV could be removed, wasn't really sure if the intent is to use it for prod or dev.

**Comment by alex-stout** [comment]
Updated to run as non-root user. I kept in `NODE_ENV = "development"` to be explicit that this is meant for development only. If this Docker configuration is deemed "production" ready, that could be changed. Thank you for the input, that helps a lot.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+USER node
+
+WORKDIR /home/node
+
+COPY --chown=node:node . .
+
+RUN cp ./api/package*.json ./ && npm install
+
+COPY . .
```
Comment (battlecow) [inline Dockerfile]: This can be removed then since we are copying everything already on line 9 and ensuring node owns the files on copy.

Comment (alex-stout) [reply Dockerfile]: Removed those lines.

File: Dockerfile [hunk 2]
```diff
@@ -8,10 +8,6 @@ WORKDIR /home/node
 
 COPY --chown=node:node . .
 
-RUN cp ./api/package*.json ./ && npm install
```
Comment (battlecow) [inline Dockerfile]: Sorry line 11 should stay so the image can be run by default.

**Review by battlecow — APPROVED** [review]
LGTM

**Review by jacobschultz — APPROVED** [review]
Looks good!

**Pull request merged**